read-fonts/src/tables/truetype/deltas.rs

```diff
@@ -0,0 +1,350 @@
+use core::ops::RangeInclusive;
+
+use crate::{
+    tables::glyf::{PointFlags, PointMarker},
+    tables::gvar::{Gvar, TupleVariation},
+    types::{F2Dot14, Fixed, GlyphId, Point},
+    ReadError,
+};
+
+pub type Delta = Point<Fixed>;
+
+/// Compute a set of deltas for the component offsets of a composite glyph.
+///
+/// Interpolation is meaningless for component offsets so this is a
+/// specialized function that skips the expensive bits.
+pub fn composite_glyph(
+    gvar: &Gvar,
+    glyph_id: GlyphId,
+    coords: &[F2Dot14],
+    deltas: &mut [Delta],
+) -> Result<(), ReadError> {
+    compute_deltas_for_glyph(gvar, glyph_id, coords, deltas, |scalar, tuple, deltas| {
+        for tuple_delta in tuple.deltas() {
+            let ix = tuple_delta.position as usize;
+            if let Some(delta) = deltas.get_mut(ix) {
+                *delta += tuple_delta.apply_scalar(scalar);
+            }
+        }
+        Ok(())
+    })?;
+    Ok(())
+}
+
+pub struct SimpleGlyph<'a> {
+    pub points: &'a [Point<i32>],
+    pub flags: &'a mut [PointFlags],
+    pub contours: &'a [u16],
+}
+
+/// Compute a set of deltas for the points in a simple glyph.
+///
+/// This function will use interpolation to infer missing deltas for tuples
+/// that contain sparse sets. The `working_points` buffer is temporary storage
+/// used for this and the length must be >= glyph.points.len().
```

Comment:
shouldn't we have a guard clause to check this and error if it's not true?

Reply:
Yes, good catch! I’ll remove the potential panics.

Reply:
Fixed